Fakebook.Posts/Fakebook.Posts.DataAccess/Repositories/PostsRepository.cs

```diff
@@ -20,32 +20,32 @@ public PostsRepository(FakebookPostsContext context)
             _context = context;
         }
 
-        public async Task<IEnumerable<Post>> NewsfeedAsync(string email, int count)
+        public async Task<IEnumerable<Post>> NewsfeedAsync(ICollection<string> followingemail, int maxPost = 50)
         {
-            var posts = await _context.Posts.FromSqlInterpolated(
-                $"SELECT * FROM ( SELECT *, ROW_NUMBER() OVER ( PARTITION BY \"UserEmail\" ORDER BY \"CreatedAt\" DESC ) AS \"RowNum\" FROM \"Fakebook\".\"Post\" WHERE \"UserEmail\" = {email} OR \"UserEmail\" IN ( SELECT \"FollowedEmail\" FROM \"Fakebook\".\"UserFollows\" WHERE \"FollowerEmail\" = {email} ) ) AS \"RecentPosts\" WHERE \"RecentPosts\".\"RowNum\" <= {count}"
-            ).ToListAsync();
-            return posts.Select(p => p.ToDomain());
+            if (followingemail != null)
+            {
+                List<Domain.Models.Post> posts = new List<Domain.Models.Post>();
+
+
+                var recentpost = await _context.Posts.OrderByDescending(t => t.CreatedAt).Take(maxPost).ToListAsync();
+
+                foreach (var item in recentpost)
+                {
+
+                    if (followingemail.Contains(item.UserEmail))
+                    {
+                        posts.Add(item.ToDomain());
+                    }
+                }
```

Comment:
the logic is off. the limit of 50 should apply after we filter by email, not before. if it's before, then we might come up with 0 posts purely because all the matching posts are a bit old.

Reply:
returning post.take(maxpost) to fix my logic issue.

Reply:
loading every post from the database doesn't scale well. can't we work the EF logic so that the filtering and limiting is handled by SQL? ideally it would be with one query, but maybe you need to have a setup query where you get the IDs of all the emails in question.

Reply:
rewrote the logic to filter beforehand. Its not going to grab every item from the database, just the items based on the users following.